service/resource/cloudformation/desired_test.go

```diff
@@ -0,0 +1,66 @@
+package cloudformation
+
+import (
+	"context"
+	"testing"
+
+	awsCF "github.com/aws/aws-sdk-go/service/cloudformation"
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/clustertpr"
+	"github.com/giantswarm/clustertpr/spec"
+	"github.com/giantswarm/micrologger/microloggertest"
+)
+
+func Test_Resource_Cloudformation_GetDesiredState(t *testing.T) {
+	testCases := []struct {
+		obj          interface{}
+		expectedName string
+		description  string
+	}{
+		{
+			description: "CloudFormation gets name from custom object",
+			obj: &awstpr.CustomObject{
+				Spec: awstpr.Spec{
+					Cluster: clustertpr.Spec{
+						Cluster: spec.Cluster{
+							ID: "5xchu",
+						},
+					},
+				},
+			},
+			expectedName: "cluster-main-stack-5xchu",
```

Comment:
Should the test use the new name format?

Reply:
Of course, but i'm not sure if we should be testing the name here at all, we have already in place a unit test for the new key method that gives the name, maybe test that the key method is called? We can discuss in the new PR.